### PR TITLE
Have deployment view output in table

### PR DESCRIPTION
**Pull request by jspspike**
Have new view for `wrangler deployment view` command
Example output
```
 ⛅️ wrangler 2.12.0
--------------------
🚧`wrangler deployments` is a beta command. Please report any issues to https://github.com/cloudflare/workers-sdk/issues/new/choose


Deployment ID: 07d7143d-0284-427e-ba22-2d5e6e91b479
Created on:    2023-03-02T21:05:15.622446Z
Author:        jspspike@gmail.com
Source:        Upload from Wrangler 🤠
------------------------------------------------------------
Author ID:          e5a3ca86e08fb0940d3a05691310bb42
Usage Model:        bundled
Handlers:           fetch
Compatibility Date: 2022-10-03
--------------------------bindings--------------------------
[[r2_buckets]]
binding = "MY_BUCKET"
bucket_name = "testr2"

[[kv_namespaces]]
id = "79300c6d17eb4180a07270f450efe53f"
binding = "yeee"

---------------------------script---------------------------

(() => {
  // index.js
  addEventListener("fetch", (event) => {
    event.respondWith(handleRequest(event.request));
  });
  async function handleRequest(request) {
    return new Response(JSON.stringify("Hello r2"), {
      headers: { "content-type": "text/plain" }
    });
  }
})();
//# sourceMappingURL=index.js.map
```

Author has included the following, where applicable:

- [x] Tests
- [x] Changeset

Reviewer has performed the following, where applicable:

- [ ] Checked for inclusion of relevant tests
- [ ] Checked for inclusion of a relevant changeset
- [ ] Checked for creation of associated docs updates
- [ ] Manually pulled down the changes and spot-tested

Fixes # [insert issue number].

### CODE REVIEW

File: .changeset/brown-wasps-fail.md [hunk 1]
```diff
@@ -14,27 +14,36 @@ in a developer's local setup.
...
+binding = "MY_BUCKET"
+bucket_name = "testr2"
+
+[[kv_namespaces]]
+id = "79300c6d17eb4180a07270f450efe53f"
+binding = "yeee"
+
+---------------------------script---------------------------
+
+(() => {
```
Comment (JacobMGEvans) [inline .changeset/brown-wasps-fail.md:37]: Lets use Module Workers even in examples like here.

File: packages/wrangler/src/deployments.ts [hunk 2]
```diff
@@ -81,7 +91,7 @@ export async function deployments(
 Deployment ID: ${versions.id}
 Created on:    ${versions.metadata.created_on}
 Author:        ${versions.metadata.author_email}
-Source:       ${triggerStr}`;
+Source:        ${triggerStr}`;
```
Comment (JacobMGEvans) [inline packages/wrangler/src/deployments.ts:94]: Nice catch!

File: packages/wrangler/src/deployments.ts [hunk 3]
```diff
@@ -221,26 +231,46 @@ export async function viewDeployment(
...
-
-	logger.log(flatObj);
+	const triggerStr = deploymentDetails.annotations?.["workers/triggered_by"]
+		? `${formatTrigger(
+				deploymentDetails.annotations["workers/triggered_by"]
+		  )} from ${formatSource(deploymentDetails.metadata.source)}`
+		: `${formatSource(deploymentDetails.metadata.source)}`;
+
+	const rollbackStr = deploymentDetails.annotations
+		? ["workers/rollback_from"]
```
Comment (JacobMGEvans) [inline packages/wrangler/src/deployments.ts:241]: I think you meant to do a optional chaining, which can be done with dot notation `?.` however not with bracket notation `objection['bracketNotation']`
```suggestion
	const rollbackStr = deploymentDetails.annotations["workers/rollback_from"]
```

**Review by JacobMGEvans — CHANGES_REQUESTED** [review]
The PR looks like a great refactor and polishing up of the output!! There are a couple things to fix and a few linter issues.

**Comment by Skye-31** [comment]
It might be a good idea to hide the contents of the script behind a flag or something similar - if someone has a large worker, such as a 5mb one, this would flood their terminal

**Comment by JacobMGEvans** [comment]
> It might be a good idea to hide the contents of the script behind a flag or something similar - if someone has a large worker, such as a 5mb one, this would flood their terminal
> 
> Could also truncate the contents of the script if a flag isn't desired

@admah Interesting thought.

**Review by JacobMGEvans — APPROVED** [review]

File: packages/wrangler/src/index.ts [hunk 4]
```diff
@@ -625,6 +625,11 @@ export function createCLIParser(argv: string[]) {
 								type: "string",
 								demandOption: true,
 							})
+							.option("content", {
+								describe: "Show script content for given deployment ID",
+								type: "boolean",
+								default: false,
+							})
```
Comment (Skye-31) [inline packages/wrangler/src/index.ts:632]: Thank you!

Comment (admah) [reply packages/wrangler/src/index.ts:632]: @Skye-31 thank you for bringing the issue up!

**Pull request merged**